bootstrap/kubeadm/api/v1alpha3/kubeadmbootstrapconfig_types.go

```diff
@@ -175,6 +175,98 @@ type File struct {
 	Content string `json:"content"`
 }
 
+// FileSource is a union of all possible external source types for file data.
+// Only one field may be populated in any given instance. Developers adding new
+// sources of data for target systems should add them here.
+type FileSource struct {
+	// Raw represents raw user-provided string content to populate this file.
+	Raw *File `json:"raw,omitempty"`
+
+	// Secret represents a secret that should populate this file.
+	// +optional
+	Secret *SecretFileSource `json:"secret,omitempty"`
+
+	// ConfigMap represents a configMap that should populate this volume
+	// +optional
+	ConfigMap *ConfigMapFileSource `json:"configMap,omitempty"`
```

Comment:
Is there a use case for ConfigMaps? 
I'm debated betweek ok, if we have secrets then also ConfigMaps are ok, and let's keep this simple...

Reply:
removed until someone asks for it